consensus/hotstuff/voteaggregator/vote_aggregator.go

```diff
@@ -185,6 +211,37 @@ func (va *VoteAggregator) processQueuedVote(vote *model.Vote) error {
 	return nil
 }
 
+// processQueuedBlock performs actual processing of queued block proposals, this method is called from multiple
+// concurrent goroutines.
+func (va *VoteAggregator) processQueuedBlock(block *model.Proposal) error {
+	// check if the block is for a view that has already been pruned (and is thus stale)
+	if block.Block.View < va.lowestRetainedView.Value() {
+		return nil
+	}
+
+	collector, created, err := va.collectors.GetOrCreateCollector(block.Block.View)
+	if err != nil {
+		if mempool.IsBelowPrunedThresholdError(err) {
+			return nil
+		}
+		return fmt.Errorf("could not get or create collector for block %v: %w", block.Block.BlockID, err)
+	}
+
+	if created {
+		va.log.Info().
+			Uint64("view", block.Block.View).
+			Hex("block_id", block.Block.BlockID[:]).
+			Msg("vote collector is created by processing block")
+	}
+
+	err = collector.ProcessBlock(block)
+	if err != nil {
```

Comment:
I've updated error handling to throw exception, strictly speaking receiving `model.InvalidBlockError` means we have a serious bug somewhere since it has to receive only fully validated blocks.